src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -37997,6 +37997,204 @@ static void Main()
                 Diagnostic(ErrorCode.ERR_NoTypeDef, "[x, ..y]").WithArguments("MyCollectionA<>", $"{assemblyA}, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null").WithLocation(7, 35));
         }
 
+        [WorkItem("https://github.com/dotnet/roslyn/issues/72898")]
+        [Fact]
+        public void Nullable_ConditionalOperator_Error()
+        {
+            string sourceA = """
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                using System.Runtime.CompilerServices;
+                [CollectionBuilder(typeof(MyCollectionBuilder), "Create")]
+                struct MyCollection<T> : IEnumerable<T>
+                {
+                    IEnumerator<T> IEnumerable<T>.GetEnumerator() => null;
+                    IEnumerator IEnumerable.GetEnumerator() => null;
+                }
+                class MyCollectionBuilder
+                {
+                    public static MyCollection<T> Create<T>(ReadOnlySpan<T> items) => default;
+                }
+                """;
+            string sourceB = """
+                #nullable enable
+                using System.Collections.Generic;
+                class Program
+                {
+                    static IEnumerable<object> F(bool b, MyCollection<object> x, object y)
+                    {
+                        return b ? x : [y);
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation([sourceB, sourceA], targetFramework: TargetFramework.Net80);
+            comp.VerifyEmitDiagnostics(
+                // (7,26): error CS1003: Syntax error, ',' expected
+                //         return b ? x : [y);
+                Diagnostic(ErrorCode.ERR_SyntaxError, ")").WithArguments(",").WithLocation(7, 26),
+                // (7,27): error CS1003: Syntax error, ']' expected
+                //         return b ? x : [y);
+                Diagnostic(ErrorCode.ERR_SyntaxError, ";").WithArguments("]").WithLocation(7, 27));
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var expr = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Last();
+            Assert.Equal("y", expr.ToString());
+            _ = model.GetSymbolInfo(expr);
```

Comment:
Consider verifying the result since the change to `BoundConditionalOperator.NaturalTypeOpt` is surfaced through `.Type` from this API